game.js

```diff
@@ -0,0 +1,79 @@
+var matchIt = 0;
+var arr = [];
+var score = 0;
+
+function gotAMatch() {
+	setTimeout(function(){
+		$('.selected').each(function(){
+			$(this).removeClass('selected');
+			$(this).addClass('matched');
+		});
+	},300);
+	score++;
+	$('#score').html(score);
+}
+
+function createRandomArray() {
+	//Create an array of randomly shuffled numbers up to the total number of cards
+	function shuffledArray(array) {
+	  var currentIndex = array.length, temporaryValue, randomIndex ;
+	  // While there remain elements to shuffle...
+	  while (0 !== currentIndex) {
+		// Pick a remaining element...
+		randomIndex = Math.floor(Math.random() * currentIndex);
+		currentIndex -= 1;
+		// And swap it with the current element.
+		temporaryValue = array[currentIndex];
+		array[currentIndex] = array[randomIndex];
+		array[randomIndex] = temporaryValue;
+	  }
+	  return array;
+	}
```

Comment:
Avoid creating named functions in other named functions, because the inner function needs to be re-created each time the outer one is called. Also, it makes the outer function harder to follow.